draft-irtf-cfrg-aead-limits.md

```diff
@@ -344,77 +362,99 @@ v * 2^64 + (2l * (v + q))^2 <= p * 2^128
-~~~
+### Authenticated Encryption Security Limit
 
-This implies the following limit:
+<!--
+    From {{GCM-MU2}} Theorem 4.3.
+
+    Let:
+        - #blocks encrypted/verified overall:   \sigma = (q + v) * l
+        - worst-case  o (offline work), q+v, \sigma <= 2^95
+          (Theorem 4.3 requires q <= 2^(1-e)r ; this yields e >= 0.0104, hence
+          d = 1,5/e -1 <= 143 <= 2^8.)
 
+    We can simplify as follows:
+        - Note: Last term is 2^-48; hence any other term <= 2^-50 is negligible.
+        - 1st term (../2^k):  roughly <= 2^8 * (o + q+v + \sigma) / 2^k
+           roughly <= (o + (q+v)*l) / 2^(k-8)
+          This is negligible for k = 256.
+          For k = 128, it is negligible if o, (q+v)*l <= 2^70.
+          For o <= 2^70 and B >= 2^8, it is dominated by the 2nd term;
+            we assume that and hence omit the 1st term.
+        - 2nd term (../2^n):  roughly  = \sigma*B/2^127
+        - 3rd term (../2^2n):  <= 2^-160, negligible.
+        - 4th term (../2^(k+n)):  roughly <= (\sigma^2 + 2o(q+v)) / 2^256
+          <= 2^-64, negligible.
+        - 5th term (2^(-r/2)):  = 2^48
+-->
 ~~~
-v + q <= sqrt(p * u * 2^128) / l
+AE <= (q+v)*l*B / 2^127 + 1/2^48
 ~~~
 
-### Integrity Limit
+This implies the following limit (where the adversary attack probability `p` is
+always `>= 2^-48`):
 
-<!-- From Bad_8 advantage contribution to the inequality from 4.3 in {{GCM-MU}},
-  assuming \sigma = (v+e)*l -->
 ~~~
-CA <= (1 / 2^1024) + ((2 * (v + q)) / 2^256)
-        + ((2 * o * (v + q)) / 2^(k + 128))
-        + (128 * ((v + q) + ((v + q) * l)) / 2^k)
+q + v <= p * 2^127 / (l * B)
 ~~~
 
-When k = 128, the last term in this inequality dominates. Thus, we can simplify
-this to:
+### Confidentiality Limit
+
+<!--
+    From {{GCM-MU2}} Theorem 4.3,
+    substracting terms for Pr[Bad_7] and Pr[Bad_8],
+    and applying simplifications as above (note there a no verification queries),
+    we obtain:
 
-~~~
-CA <= (128 * ((v + q) + ((v + q) * l)) / 2^128)
-~~~
+    Adv^{mu-ae w/o INT}_RCAU <=
+        2^8 * (o + q) / 2^k   +  \sigma*B/2^127  +  2^48
 
-This implies the following limit:
+    For o <= 2^70 and any B, the 1st term is dominated by the 2nd term;
+    we assume that and hence again omit the 1st term.
+-->
+
+The confidentiality advantage is essentially dominated by the same terms as
+the AE advantage:
 
 ~~~
-v + q <= (p * 2^128) / (128 * (l + 1))
+CA <= q*l*B / 2^127 + 1/2^48
 ~~~
 
-When k = 256, the second and fourth terms in the CA inequality dominate. Thus, we
-can simplify this to:
+This implies the following limit (where the adversary attack probability `p` is
+always `>= 2^-48`):
 
 ~~~
-CA <= ((2 * (v + q)) / 2^256)
-        + (128 * ((v + q) + ((v + q) * l)) / 2^256)
+q <= p * 2^127 / (l * B)
 ~~~
 
-This implies the following limit:
+### Integrity Limit
+
+There is currently no dedicated multi-user integrity bound for
+AEAD_AES_128_GCM and AEAD_AES_256_GCM. The AE limit above applies as
+integrity limit.
```

Comment:
Is this what we want to say, or do we want to instead say that the AE limit can be used to derive an integrity limit based on a chosen `p` and allocation of the limit between `q` and `v`?

Reply:
Good point, I added a repetition of these explicit limits in 5d3cc7b.